src/components/vote-form.js

```diff
@@ -1,14 +1,89 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import axios from "axios";
 import styled from "styled-components";
+import CandidateForm from "./candidate-form";
 
 export default function VoteForm() {
-  return <Wrapper>안녕 나는 투표 폼!</Wrapper>;
+  const [candidateList, setCandidateList] = useState(null);
+
+  // useEffect는 렌더링 될때마다 실행, [] 인자에는 바뀔 떄마다 렌더링할 변수가 들어간다
+  // candidates의 값이 바뀔 때마다 렌더링한다.
+  useEffect(() => {
+    takeCandidateList();
+  }, [candidateList]);
+
+  // axois에서 후보자 명단 받아온다.
+
+  const takeCandidateList = async () => {
+    const data = await axios
+      .get(process.env.API_HOST + "/candidates/", {
+        params: {},
+      })
+      .then(function (response) {
+        return response.data;
+      })
+      .catch(function (error) {
+        console.log(error);
+      })
+      .finally(function () {
+        // always executed
+      });
+    setCandidateList(data);
+  };
+  // candidates값이 null이 아니면
+
+  return (
+    <Wrapper>
+      <Title1>
+        <RedText>프론트앤드 인기쟁이</RedText>는 누구?
+      </Title1>
+      <Title2>CEOS 프론트엔드 개발자 인기 순위 및 투표 창입니다.</Title2>
+      {candidateList !== null && (
+        <CandidateListWrapper>
+          {candidateList
+            .sort((a, b) => {
+              return b.voteCount - a.voteCount;
+            })
+            .map((candidate) => (
+              <CandidateForm
+                key={candidate._id}
+                id={candidate._id}
+                rank={candidateList.indexOf(candidate) + 1}
+                name={candidate.name}
+                voteCount={candidate.voteCount}
+              />
+            ))}
+        </CandidateListWrapper>
+      )}
```

Comment:
1. candidateList !== null && 대신에 candidateList && 를 해도 똑같이 작동해요.

2. candidateList가 없으면 어짜피 안에 아무것도 안 나오기 때문에 CandidateListWrapper를 candidateList체크하기전에 감싸주는 게 가독성에 좋은 거 같아요!

3. map콜백의 두번째 인자로 배열의 인덱스 값을 받을 수 있습니다. candidateList.indexOf(candidate) 이렇게 index를 구하지 않아도 됩니다~
  
```suggestion
      <CandidateListWrapper >
        {candidateList &&
          candidateList
            .sort((a, b) => b.voteCount - a.voteCount)
            .map((candidate, index) => {
              const { _id : id, name, voteCount } = candidate;
              return (
               <CandidateForm
                key={id}
                rank={index + 1}
                 {...{name,voteCount,id}}
                />
              );
            })}
      </CandidateListWrapper>
```